src/shared/components/VideoPlayer/ControlsIndicator.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState } from 'react'
+import { CSSTransition } from 'react-transition-group'
+import { VideoJsPlayer } from 'video.js'
+
+import {
+  SvgPlayerBackwardFiveSec,
+  SvgPlayerBackwardTenSec,
+  SvgPlayerForwardFiveSec,
+  SvgPlayerForwardTenSec,
+  SvgPlayerPause,
+  SvgPlayerPlay,
+  SvgPlayerSoundHalf,
+  SvgPlayerSoundOff,
+  SvgPlayerSoundOn,
+} from '@/shared/icons'
+
+import {
+  ControlsIndicatorIconWrapper,
+  ControlsIndicatorTooltip,
+  ControlsIndicatorTransitions,
+  ControlsIndicatorWrapper,
+} from './ControlsIndicator.style'
+import { CustomVideojsEvents } from './videoJsPlayer'
+
+import { Text } from '../Text'
+
+type VideoEvent = CustomVideojsEvents | null
+
+type EventState = {
+  type: VideoEvent
+  description: string | null
+  icon: React.ReactNode | null
+  isVisible: boolean
+}
+
+type ControlsIndicatorProps = {
+  player: VideoJsPlayer | null
+}
+
+export const ControlsIndicator: React.FC<ControlsIndicatorProps> = ({ player }) => {
+  const [indicator, setIndicator] = useState<EventState | null>(null)
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    let timeout: NodeJS.Timeout
```

Comment:
We're not in nodejs environment, let's not use this type. `setTimeout` returns a regular `number`

Reply:
hmm, number doesn't work.
![image](https://user-images.githubusercontent.com/51168865/126621989-b43d4abe-a163-4fec-81b5-6244088b6471.png)
![image](https://user-images.githubusercontent.com/51168865/126622024-3ff2367e-385c-44cd-9e21-29d296f2e3a8.png)
I'll just use `ReturnType<typeof setTimeout>`

Reply:
Oh I see, to make this work I would have to write `window.setTimeout`, will stick to ReturnType

Reply:
Actually, thinking about this now, we actually should use `window.setTimeout`, `setTimeout` isn't a global function and it only works because by default CRA webpack will polyfill node globals